app/survey_config/survey_config.py

```diff
@@ -40,13 +42,18 @@ class SurveyConfig:
     footer_legal_links: Optional[Iterable[Mapping]] = None
     survey_title: Optional[LazyString] = None
     design_system_theme: Optional[str] = None
-    data_layer: Iterable[Union[Mapping]] = field(default_factory=list, compare=False)
+    data_layer: list[dict[str, Union[str, bool]]] = field(
+        default_factory=list, compare=False
+    )
     sign_out_button_text: str = lazy_gettext("Save and exit survey")
     contact_us_url: str = field(init=False)
     cookie_settings_url: str = field(init=False)
     privacy_and_data_protection_url: str = field(init=False)
 
     def __post_init__(self):
+        if current_user and get_metadata(current_user):
+            if tx_id := get_metadata(current_user).get("tx_id"):
+                self.data_layer.append({"tx_id": tx_id})
```

Comment:
App context should be avoided in these classes. Data layer will have to be done similarly to `get_service_links`. We should pass the tx_id in to avoid needed the app context. Same concept as the service links.

Reply:
Done this as you suggested, a lot of refactoring involved. Thanks!